buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -159,25 +194,38 @@ ByteBuf allocate(int size, int maxCapacity) {
 
     private AdaptiveByteBuf allocate(int size, int maxCapacity, Thread currentThread, AdaptiveByteBuf buf) {
         long threadId = currentThread.getId();
-        int sizeBucket = AllocationStatistics.sizeBucket(size); // Compute outside of Magazine lock for better ILP.
+        int sizeBucket = AllocationStatistics.sizeBucket(size);
+        FastThreadLocal<Magazine> fastThreadLocalMagazines = this.fastThreadLocalMagazines;
+        if (fastThreadLocalMagazines != null && currentThread instanceof FastThreadLocalThread) {
+            // TODO this is opinionated: ignoring on purpose non FastThreadLocalThread(s):
+            //      we could add ThreadExecutorMap.currentExecutor() != null to make sure this is an event loop thread
+            //      because there are frameworks which uses FastThreadLocalThread(s) for their blocking pool threads!
+            Magazine mag = fastThreadLocalMagazines.get();
+            return mag.allocate(size, sizeBucket, maxCapacity, buf);
+        }
+        // Compute outside of Magazine lock for better ILP.
```

Comment:
This comment got divorced from `int sizeBucket = AllocationStatistics.sizeBucket(size);` on line 197.